scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1340,6 +1340,16 @@ class Router(formatOps: FormatOps) {
           Split(Space, 0, policy = singleLine),
           Split(Newline, 1).withPolicy(penalizeNewlineByNesting(cond, arrow))
         )
+      case FormatToken(_, dot @ T.Dot(), _) =>
+        // allow keeping a newline (and indent) within a select chain
+        val noNewLine = !style.activeForEdition_2019_11 ||
+          newlines == 0 || rightOwner.isNot[Term.Select]
+        val newlinePenalty = if (noNewLine) 0 else 10 + treeDepth(rightOwner)
+        Seq(
```

Comment:
FWIW, doubling the search state here is likely to introduce new "search state exploded" errors for programs that formatted fine before.